src/Address.php

```diff
@@ -2,255 +2,32 @@
-            $address->houseExtension = $data['house_extension'];
-        }
-
-        if (isset($data['postal_code'])) {
-            $address->postalCode = $data['postal_code'];
-        }
-
-        if (isset($data['city'])) {
-            $address->city = $data['city'];
-        }
-
-        if (isset($data['region'])) {
-            $address->region = $data['region'];
-        }
-
-        if (isset($data['phone'])) {
-            $address->phone = $data['phone'];
-        }
-
-        if (isset($data['country'])) {
-            $address->country = $data['country'];
-        }
-
-        return $address;
-    }
-
-    /**
-     * @return string
-     */
-    public function getFamilyName()
-    {
-        return $this->familyName;
-    }
-
-    /**
-     * @return string
-     */
-    public function getGivenName()
-    {
-        return $this->givenName;
-    }
-
-    /**
-     * @return string
-     */
-    public function getEmail()
-    {
-        return $this->email;
-    }
-
-    /**
-     * @return string
-     */
-    public function getTitle()
-    {
-        return $this->title;
-    }
-
-    /**
-     * @return string
-     */
-    public function getStreetAddress()
-    {
-        return $this->streetAddress;
-    }
-
-    /**
-     * @return string
-     */
-    public function getStreetAddress2()
-    {
-        return $this->streetAddress2;
-    }
-
-    /**
-     * @return string
-     */
-    public function getStreetName()
-    {
-        return $this->streetName;
-    }
-
-    /**
-     * @return string
-     */
-    public function getStreetNumber()
-    {
-        return $this->streetNumber;
-    }
-
-    /**
-     * @return string
-     */
-    public function getHouseExtension()
-    {
-        return $this->houseExtension;
-    }
-
-    /**
-     * @return string
-     */
-    public function getPostalCode()
-    {
-        return $this->postalCode;
-    }
-
-    /**
-     * @return string
-     */
-    public function getCity()
-    {
-        return $this->city;
-    }
-
-    /**
-     * @return string
-     */
-    public function getRegion()
-    {
-        return $this->region;
-    }
-
-    /**
-     * @return string
-     */
-    public function getPhone()
-    {
-        return $this->phone;
-    }
-
-    /**
-     * @return string
-     */
-    public function getCountry()
-    {
-        return $this->country;
+        $defaults = [
+            'family_name' => null,
+            'given_name' => null,
+            'email' => null,
+            'title' => null,
+            'street_address' => null,
+            'street_address2' => null,
+            'street_name' => null,
+            'house_extension' => null,
+            'street_number' => null,
+            'postal_code' => null,
+            'city' => null,
+            'region' => null,
+            'phone' => null,
+            'country' => null,
+        ];
+
+        return new self(array_merge($defaults, $data));
```

Comment:
Did you leave out `array_intersect_key` intentionally? Since this will allow additional undefined parameters from data